typescript/sdk/src/token/schemas.ts

```diff
@@ -71,7 +95,72 @@ export const WarpRouteDeployConfigSchema = z
     const entries = Object.entries(configMap);
     return (
       entries.some(
-        ([_, config]) => isCollateralConfig(config) || isNativeConfig(config),
+        ([_, config]) =>
+          isCollateralConfig(config) ||
+          isCollateralRebaseConfig(config) ||
+          isNativeConfig(config),
       ) || entries.every(([_, config]) => isTokenMetadata(config))
     );
-  }, `Config must include Native or Collateral OR all synthetics must define token metadata`);
+  }, WarpRouteDeployConfigSchemaErrors.NO_SYNTHETIC_ONLY)
+  .transform((warpRouteDeployConfig, ctx) => {
+    const hasCollateralRebase = Object.entries(warpRouteDeployConfig).find(
+      ([_, config]) => isCollateralRebaseConfig(config),
+    );
+    if (!hasCollateralRebase) return warpRouteDeployConfig;
+
+    if (!isCollateralRebasePairedCorrectly(warpRouteDeployConfig)) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: WarpRouteDeployConfigSchemaErrors.ONLY_SYNTHETIC_REBASE,
+      });
+
+      // This is a special symbol you can use to
+      // return early from the transform function.
+      // It has type `never` so it does not affect the
+      // inferred return type.
+      return z.NEVER;
+    }
+
+    const modifiedConfigMap = resolveCollateralChainName(warpRouteDeployConfig);
+    return modifiedConfigMap;
+  });
+
+function isCollateralRebasePairedCorrectly(
```

Comment:
I feel like theres a decent amount of branching somewhat duped across these functions
might be less verbose without the helpers

Reply:
removed one of the helpers